### ISSUE TITLE
Support evm tx in interchain-account host chain

### ISSUE THREAD

**Issue by yihuang** [tharsis/ethermint#1093]
__Proposal:__ Support EVM tx in interchain-account host chain

__Current behavior:__

In general, EVM module relies on a custom ante handler pipeline to work, but the ICA host module will run the msgs under a default ante handler, there'll be issues with that, specifically:
- Gas Meter Reset
  - `EthereumTx` expects an infinite gas meter set in advance and reset the gas meter with the gas calculated by EVM, that don't work well with the default msg executor of the interchain-account host module.
- Gas Fee Payment and relayer incentivization.
  - we need to have some kind of relayer incentivization first because the cost of evm tx varies.
- JSON-RPCs
  - it emits the same events, so it can be found by json-rpc apis similarly to the batch tx.
    - [the current pending PR](https://github.com/tharsis/ethermint/pull/1062) tries to move some events to the ante handler, we should keep a copy in the msg handler too then.
  - it breaks client's assumption that the fee is always paid by the tx sender, it's relayer instead.

__Desired behavior:__ make evm tx works smoothly with interchain-account

__Use case:__ support interchain-account host role

Possible solutions:
- make a custom msg handler which is implemented with interchain-account in mind?

**Comment by tomtau** [comment]
maybe one "manual" alternative is directly constructing those messages in precompile: https://github.com/crypto-org-chain/cronos/issues/483 ? it won't be "smooth" though, I guess: EVM will see invocation via interchain accounts as from EOA, but taking IBC messages from a precompile and interpreting them would not leverage much of the ibc-go code and there'll be an indirection

**Comment by yihuang** [comment]
> maybe one "manual" alternative is directly constructing those messages in precompile: [crypto-org-chain/cronos#483](https://github.com/crypto-org-chain/cronos/issues/483) ? it won't be "smooth" though, I guess: EVM will see invocation via interchain accounts as from EOA, but taking IBC messages from a precompile and interpreting them would not leverage much of the ibc-go code and there'll be an indirection

do you mean some custom evm to evm interchain-accounts implementation?

**Comment by yihuang** [comment]
I'm thinking about making a cosmos-sdk friendly evm msg type, sth like:
```protobuf
message EvmMessage {
  sender string
  to string
  value *big.Int
  data []byte
}
```
it'll leave the auth/fee stuff to cosmos-sdk, works just like a normal cosmos-sdk msg.

**Comment by tomtau** [comment]
> do you mean some custom evm to evm interchain-accounts implementation?

Yes... one potential advantage is that one may want to restrict contract addresses that it can interact with (similarly to how one can restrict SDK message types for ICA).

> it'll leave the auth/fee stuff to cosmos-sdk, (...)

as long as it doesn't bypass gas accounting (i.e. being able to pay significantly less than when going through the Ethermint's ante handler), it looks feasible
